File: java/estructuras_de_datos/buscar_contacto.java
package estructuras_de_datos;
import java.util.ArrayList;
import java.util.Scanner;
public class buscar_contacto {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<String> contactos = new ArrayList<>();
        contactos.add("Ana Gómez - 3001112233");
        contactos.add("Luis Pérez - 3012223344");
        contactos.add("Sofía Martínez - 3023334455");
        System.out.print("Ingrese el nombre del contacto a buscar: ");
        String nombreBuscar = scanner.nextLine();
        boolean encontrado = false;
        for (String contacto : contactos) {
            if (contacto.toLowerCase().contains(nombreBuscar.toLowerCase())) {
                System.out.println("Contacto encontrado: " + contacto);
                encontrado = true;
            }
        }
        if (!encontrado) {
            System.out.println("Contacto no encontrado.");
        }
        scanner.close();
    }
}

File: visual_basic/arreglos_y_listas/buscar_contacto/buscar_contacto.vb
Imports System
Module buscar_contacto
    Sub Main(args As String())
        Dim contactos As New List(Of String) From {
            "Juan",
            "Maria",
            "Carlos",
            "Ana"
        }
        Console.WriteLine("=== BUSCAR CONTACTO ===")
        Console.WriteLine("Ingrese el nombre a buscar:")
        Dim nombreBuscar As String = Console.ReadLine()
        Dim encontrado As Boolean = False
        For Each contacto As String In contactos
            If contacto.ToLower() = nombreBuscar.ToLower() Then
                encontrado = True
                Exit For
            End If
        Next
        If encontrado Then
            Console.WriteLine("✅ Contacto encontrado")
        Else
            Console.WriteLine("❌ Contacto no encontrado")
        End If
        Console.ReadKey()
    End Sub
End Module